pallets/delegation/src/lib.rs

```diff
@@ -260,50 +268,21 @@ decl_module! {
 		pub fn revoke_delegation(origin, delegation_id: T::DelegationNodeId, max_depth: u64) -> DispatchResult {
 			// origin of the transaction needs to be a signed sender account
 			let sender = ensure_signed(origin)?;
-			// check if delegation node exists
-			if !<Delegations<T>>::contains_key(delegation_id) {
-				return Self::error(Self::ERROR_DELEGATION_NOT_FOUND)
-			}
+			// check if a delegation node with the given identifier already exists
+			ensure!(<Delegations<T>>::contains_key(delegation_id), Error::<T>::DelegationNotFound);
+
 			// check if the sender of this transaction is permitted by being the
 			// owner of the delegation or of one of its parents
-			if !Self::is_delegating(&sender, &delegation_id, max_depth)? {
-				return Self::error(Self::ERROR_NOT_PERMITTED_TO_REVOKE)
-			}
+			ensure!(Self::is_delegating(&sender, &delegation_id, max_depth)?, Error::<T>::UnauthorizedRevocation);
+
 			// revoke the delegation and recursively all of its children
-			Self::revoke(&delegation_id, &sender)?;
-			Ok(())
+			Self::revoke(&delegation_id, &sender)
```

Comment:
clippy would have found it, if it wasn't inside this macro. 😞 Looking forward to frame 2.0! 😋